x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapper.java

```diff
@@ -186,12 +212,8 @@ public boolean equals(Object other) {
                 CartesianPoint o = (CartesianPoint)other;
                 oX = o.getX();
                 oY = o.getY();
-            } else if (other instanceof ParsedCartesianPoint == false) {
```

Comment:
We don't need to be equal to `ParsedCartesianPoint`s?

Reply:
If we're comparing against a `ParsedCartesianPoint`, then the earlier `instanceof CartesianPoint` branch will already have matched, so this branch is never called.

Reply:
++. I hadn't pulled it up.